### PR TITLE
fix(validators): allow empty schedule filter params

### LINKED ISSUE #1269: Schedule filter broken: empty form params fail URI validation, redirect strips query parameters
## Bug Description

The schedule filter is not working: when submitting the filter form on `schedule.php`, the page redirects back to `schedule.php` without any query parameters, effectively discarding the filter.

**Expected**: Navigating to something like `Web/schedule.php?resourceId%5B%5D=36&resourceId%5B%5D=37&RESOURCE_TYPE_ID=&maxParticipants=&...`
**Actual**: Always redirects to `Web/schedule.php` with no parameters.

Additionally, as an administrator user, the Owner and Participant fields are not visible in the filter.

## Root Cause

Identified via `git bisect`:

```
1f9a739409b9cf9596e7e6c2e894b0bf6f3ee56a is the first bad commit
fix(reservation): correct guest reservation URI validation from calendar and schedule view
```

### Problem 1: Empty form params fail validation

In `ParamsValidatorMethods::existsInURLValidator()`, the logic was changed from `$matches[1] === ''` to `$matches[1] !== ''` (commit 1f9a739409). This was correct for guest reservation validation (checking that `sid` and `rid` have values), but broke the `[EXISTS, NUMERICAL]` combined validator pattern used in `RouteParamsKeys::VIEW_SCHEDULE`.

Previously, `EXISTS` acted as an "allow empty" fallback: when a form field like `RESOURCE_TYPE_ID` was submitted empty (`RESOURCE_TYPE_ID=`), `EXISTS` returned `true` (empty match), so at least one validator in the `[EXISTS, NUMERICAL]` pair passed. After the fix, `EXISTS` returns `false` for empty values, AND `numericalValidator` also returns `false` for empty values → both validators fail → `ParamsValidator::validateOrRedirect()` redirects to `schedule.php` without params.

A subsequent fix (47860f150) removed `EXISTS` from the combined validators but didn't address the underlying issue: **empty form parameter values still fail `numericalValidator`** (the regex requires at least one digit, and `ctype_digit('')` returns `false`).

### Problem 2: `SUBMIT` button value fails boolean validation

The filter form's submit button sends `SUBMIT=Filter` (the button label), but `RouteParamsKeys::VIEW_SCHEDULE` validates SUBMIT with `ParamsValidatorKeys::BOOLEAN`, which only accepts `true` or `false`. The second form has a hidden input `<input type="hidden" name='SUBMIT' value="true" />` which would pass, but the advanced filter form's submit button does not.

## Affected Pages

- `schedule.php` (`RouteParamsKeys::VIEW_SCHEDULE`)
- Likely also `view-calendar.php` (`RouteParamsKeys::VIEW_CALENDAR`) which had the same `[EXISTS, NUMERICAL]` pattern

## Suggested Fix

In `ParamsValidator::validate()`, skip validation for params that are present in the query string but have empty values — treat them the same as absent params. An empty form field means the user didn't set a filter value, so there's nothing to validate:

```php
// In the foreach loop over defined validation rules:
if (!array_key_exists($param, $queryParams)) {
    continue; // already exists
}

// Add: skip validation for empty values (user didn't set this filter)
if ($queryParams[$param] === '') {
    continue;
}
```

For the SUBMIT issue, either:
- Remove SUBMIT from `VIEW_SCHEDULE` validators (it doesn't need security validation)
- Add a hidden `<input type="hidden" name="SUBMIT" value="true">` to the advanced filter form and remove `name` from the submit button
- Or the empty-value skip above would NOT fix this since "Filter" is non-empty — this needs a separate fix

## Steps to Reproduce

1. Log in and navigate to the Schedule page
2. Open the advanced filter
3. Select one or more resources
4. Click "Filter"
5. **Result**: Page redirects to `schedule.php` with no query parameters; filter is not applied

## Environment

- Affects all versions since commit 1f9a739409 (including after the partial fix in 47860f150)
- Confirmed via `git bisect` between a known-good ancestor and current `develop`

### PR DESCRIPTION
**Pull request by JohnVillalovos**
Add optional numeric and simpledate URI validator for schedule and calendar filter query params that may be submitted as empty strings in GET forms.

Use the new validator for the optional VIEW_SCHEDULE filter fields (user, participant, resource type, and max participants) so empty filter inputs no longer fail validation and trigger a redirect to the bare page.

Keep ParamsValidator::validate() unchanged so required guest reservation routes remain strict.

Add validator and route-level regression tests covering empty, numeric, invalid, array, and XSS-shaped inputs.

This basically restores how things were validated before:

commit 47860f150610e57c15028997099be04e7441e8f2
fix(validators): remove redundant EXISTS from combined route validators

and

commit 1f9a739409b9cf9596e7e6c2e894b0bf6f3ee56a
fix(reservation): correct guest reservation URI validation from calendar and schedule view

Closes: #1269

### CODE REVIEW

**Comment by belcirelk** [comment]
Tested and it repairs the filters. The guest mode is still functional and calendars too.

File: tests/Infrastructure/Common/URIParamValidatorTest.php [hunk 1]
```diff
@@ -235,6 +237,105 @@ public static function arrayParamRejectedProvider(): array
...
+    public static function optionalSimpleDateProvider(): array
+    {
+        return [
+            'empty string is valid' => ['/Web/schedule.php?sds=', true],
+            'single date is valid' => ['/Web/schedule.php?sds=2026-03-31', true],
+            'date list is valid' => ['/Web/schedule.php?sds=2026-03-31,2026-04-01', true],
+            'missing param rejected' => ['/Web/schedule.php?sd=2026-03-31', false],
+            'invalid date rejected' => ['/Web/schedule.php?sds=bad-date', false],
+            'array param rejected' => ['/Web/schedule.php?sds[]=', false],
+            'xss rejected' => ['/Web/schedule.php?sds=&x=%3Cscript%3E', false],
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply tests/Infrastructure/Common/URIParamValidatorTest.php:266]: This is how things were before and no reports of issue. I don't want to change things without being 100% sure this is an issue.
